src/masternodes/mn_checks.cpp

```diff
@@ -4069,14 +4089,18 @@ Res ValidateTransferDomain(const CTransaction &tx,
         return DeFiErrors::TransferDomainEVMNotEnabled();
     }
 
+    if (!IsTransferDomainEnabled(height, mnview, consensus)) {
+        return DeFiErrors::TransferDomainNotEnabled();
+    }
+
     if (height >= static_cast<uint32_t>(consensus.ChangiIntermediateHeight4)) {
         if (obj.transfers.size() < 1) {
             return DeFiErrors::TransferDomainInvalid();
         }
     }
 
     for (const auto &[src, dst] : obj.transfers) {
-        auto res = ValidateTransferDomainEdge(tx, height, coins, consensus, src, dst);
+        auto res = ValidateTransferDomainEdge(tx, mnview, height, coins, consensus, src, dst);
```

Comment:
Don't need to pass mnview. Just gather all the data in one go, say `TransferDomainLiveConfig` and pass that around instead. Keeps it cleaner, better testable.